addons/l10n_it_edi_extended/models/account_move.py

```diff
@@ -0,0 +1,99 @@
+        # comes from the relationships between those taxes
+        # {line.id: LineTaxes nametuple(
+        #     withholding:  TaxData nametuple(tax, tax_kind, base_amount_currency, tax_amount_currency)
+        #     pension_fund: TaxData nametuple(tax, tax_kind, base_amount_currency, tax_amount_currency)
+        #     vat:          TaxData nametuple(tax, tax_kind, base_amount_currency, tax_amount_currency)
+        # )}
+        full_tax_details = self._prepare_edi_tax_details()['invoice_line_tax_details']
+        line_tax_map = {}
+        no_tax_data = TaxData(no_tax, None, None, None)
+        for line, vals in full_tax_details.items():
+            line_taxes_dict = {'withholding': no_tax_data, 'pension_fund': no_tax_data, 'vat': no_tax_data}
+            for _tax_name, tax_dict in vals['tax_details'].items():
+                tax = tax_dict['tax']
+                tax_kind = self._l10n_it_edi_extended_get_tax_kind(tax)
+                line_taxes_dict[tax_kind] = TaxData(tax, tax_kind, tax_dict['base_amount_currency'], tax_dict['tax_amount_currency'])
+            line_tax_map[line.id] = LineTaxes(**line_taxes_dict)
+
+        # Data for withholding taxes, sum the amounts of the same tax from each line
+        withholdings = {}
+        for _line_id, lines_taxes in line_tax_map.items():
+            if lines_taxes.withholding.tax:
+                current = withholdings.get(lines_taxes.withholding.tax.id, False)
+                tax_amount = abs(lines_taxes.withholding.tax_amount_currency) + abs(current['amount'] if current else 0.0)
+                withholdings[lines_taxes.withholding.tax.id] = {
+                    'type': lines_taxes.withholding.tax.l10n_it_withholding_type,
+                    'reason': lines_taxes.withholding.tax.l10n_it_withholding_reason,
+                    'percentage': abs(lines_taxes.withholding.tax.amount),
+                    'amount': tax_amount,
+                }
+        # Fix the total eventually as it must be without considering ritenuta
+        document_total = template_values['document_total']
+        if withholdings:
+            document_total += sum(x['amount'] for x in withholdings.values())
+
+        # Data for pension fund taxes, sum the amounts of the same tax from each line
+        pension_funds = {}
+        for _line_id, lines_taxes in line_tax_map.items():
+            if lines_taxes.pension_fund.tax:
+                current = pension_funds.get(lines_taxes.pension_fund.tax.id, False)
+                base_amount = abs(lines_taxes.pension_fund.base_amount_currency) + abs(current['imponibilecassa'] if current else 0.0)
+                tax_amount = abs(lines_taxes.pension_fund.tax_amount_currency) + abs(current['importocontributocassa'] if current else 0.0)
+                pension_funds[lines_taxes.pension_fund.tax.id] = {
+                     'tipocassa': lines_taxes.pension_fund.tax.l10n_it_pension_fund_type,
```

Comment:
There is like a good balance we can make between what you do in qweb and Python.  We should avoid to pass a dict with just the Italian values which are exactly the same in the XML.  On the other hand we should try to avoid big computations in the QWeb, but if we can show immediately where the field comes from, it is nice.  So, we could just pass the tax and in the qweb it can be shown directly where it comes from. 
```suggestion
                     'tax': lines_taxes.pension_fund.tax,
```
<AlCassa t-esc='line['tax'].l10n_it_pension_fund_type>